### PR TITLE
provisioning: Prevent re-provisioning

### PR DESCRIPTION
**Pull request by david-hazi-arm**
After successful provisioning, the code writes a pattern into the ITS, which could be checked on the next boot.

Description
-----------

Test Steps
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by urutva — DISMISSED** [review]

File: applications/freertos_iot_libraries_tests/main.c [hunk 1]
```diff
@@ -160,25 +160,36 @@ int main( void )
...
+            LogError( ( "Device key provisioning failed [%d]\n", uxReturnValue ) );
+            LogError( ( "Device cannot connect to IoT Core. Exiting...\n" ) );
+            return EXIT_FAILURE;
+        }
+
         LogInfo( ( "Device key provisioning succeeded \n" ) );
-        status = xOtaProvisionCodeSigningKey( &xOTACodeVerifyKeyHandle, 3072 );
 
-        if( status != PSA_SUCCESS )
+        /* FIXME: Magic value */
```
Comment (jasonpcarroll) [inline applications/freertos_iot_libraries_tests/main.c:176]: Is this comment going to be addressed? I see it in each main.c.

File: applications/helpers/provisioning/dev_mode_key_provisioning.c [hunk 2]
```diff
@@ -1443,4 +1449,39 @@ int xOtaProvisionCodeSigningKey( psa_key_handle_t * pxKeyHandle,
...
+    const psa_storage_uid_t uid = FIRST_BOOT_ITS_UID;
+    uint8_t boot_pattern_in_its = 0;
+    size_t read_data_length = 0;
+
+    status = psa_its_get( uid, 0, 1, &boot_pattern_in_its,
+                          &read_data_length );
+
+    if( status != PSA_SUCCESS )
+    {
+        return 0;
```
Comment (jasonpcarroll) [inline applications/helpers/provisioning/dev_mode_key_provisioning.c:1464]: Minor comment - I notice in calls to this function you check if this returns "false" and not 0. Should this function return "false" and "true" ?

**Review by jasonpcarroll — COMMENTED** [review]
Looks fine - minor comments.

**Review by aggarg — DISMISSED** [review]

File: applications/freertos_iot_libraries_tests/main.c (hunk 1 above)
Comment (david-hazi-arm) [reply applications/freertos_iot_libraries_tests/main.c]: Removed the comment, we will fix this issue in the upcoming PRs.

File: applications/helpers/provisioning/dev_mode_key_provisioning.c (hunk 2 above)
Comment (david-hazi-arm) [reply applications/helpers/provisioning/dev_mode_key_provisioning.c:1464]: Fixed

**Review by urutva — APPROVED** [review]

**Comment by urutva** [comment]
@jasonpcarroll Can you please re-review? Thanks

**Review by aggarg — APPROVED** [review]

**Review by jasonpcarroll — APPROVED** [review]

**Comment by jasonpcarroll** [comment]
@david-hazi-arm @urutva approved

**Pull request merged**